store/sqlstore/post_store.go

```diff
@@ -942,14 +942,14 @@ func (s *SqlPostStore) GetPostsSinceForSync(options model.GetPostsSinceForSyncOp
 		Where(sq.GtOrEq{"UpdateAt": options.Since}).
 		Where(sq.Eq{"ChannelId": options.ChannelId}).
 		Limit(uint64(options.Limit)).
-		OrderBy("UpdateAt"+order, "Id")
+		OrderBy("CreateAt"+order, "DeleteAt", "Id")
```

Comment:
I've run explains for both the previous implementation and the new one, and they are identical since there are sufficient indexes to support the new ordering and filtering:

```sql
mattermost_test=# EXPLAIN ANALYSE select * from posts where updateat >= 1614935961180 and channelid = 'jdnjbpjt6jbomgk3n9y9p1n8or' order by updateat, id limit 1000;
                                                                          QUERY PLAN
-------------------------------------------------------------------------------------------------
 Limit  (cost=8.17..8.17 rows=1 width=2906) (actual time=0.086..0.152 rows=2 loops=1)
   ->  Sort  (cost=8.17..8.17 rows=1 width=2906) (actual time=0.074..0.107 rows=2 loops=1)
         Sort Key: updateat, id
         Sort Method: quicksort  Memory: 25kB
         ->  Index Scan using idx_posts_channel_id_delete_at_create_at on posts  (cost=0.14..8.16
               Index Cond: ((channelid)::text = 'jdnjbpjt6jbomgk3n9y9p1n8or'::text)
               Filter: (updateat >= '1614935961180'::bigint)
 Planning time: 0.250 ms
 Execution time: 0.214 ms
(9 rows)

mattermost_test=# EXPLAIN ANALYSE select * from posts where updateat >= 1614935961180 and channelid = 'jdnjbpjt6jbomgk3n9y9p1n8or' order by createat desc, deleteat, id limit 1000;
                                                                          QUERY PLAN
-------------------------------------------------------------------------------------------------
 Limit  (cost=8.17..8.17 rows=1 width=2906) (actual time=0.077..0.136 rows=2 loops=1)
   ->  Sort  (cost=8.17..8.17 rows=1 width=2906) (actual time=0.065..0.092 rows=2 loops=1)
         Sort Key: createat DESC, deleteat, id
         Sort Method: quicksort  Memory: 25kB
         ->  Index Scan using idx_posts_channel_id_delete_at_create_at on posts  (cost=0.14..8.16
               Index Cond: ((channelid)::text = 'jdnjbpjt6jbomgk3n9y9p1n8or'::text)
               Filter: (updateat >= '1614935961180'::bigint)
 Planning time: 0.157 ms
 Execution time: 0.197 ms
(9 rows)
```

Reply:
Nice @nronas :+1:  Out of curiosity, do you know if MySQL would choose similar strategies?

Reply:
Yeap, i think so too, assuming that the same indexes are in place, which they are as far as i can see